### PR TITLE
DA_Week11_복습과제_오연재

### PR DESCRIPTION
**Pull request by jennyohyj**

### CODE REVIEW

**Comment by yegenuine** [comment]
연재님 11주차 복습과제에 대한 피드백 드립니다!

1-3. ItemsBought과 ItemsReturned에 대해 정규화작업을 진행하라고 제시되어 있기 때문에 아래와 같은 코드를 사용하시는게 더 정확할 것 같아요. 또한 fit_transform이 아니라 fit을 사용하셔서 정규화작업이 제대로 진행되지 않은 것으로 보입니다.
CD_copy[['ItemsBought', 'ItemsReturned']] = scaler.fit_transform(CD_copy[['ItemsBought', 'ItemsReturned']])

1-5. k=6인 경우를 실수로 빼먹으신 것 같은데 결과에는 영향 없어보입니다.

1-6. 4) 군집별 평균 실루엣 계수 값을 구하는 코드는 다음과 같습니다.
cd_copy.groupby('cluster')['silhouette_coeff'].mean()

2-10, 2-11. x를 fit하고 labels_를 출력하여 확인할 때
model = db.fit(x)
label=model.labels_
실행 후 label을 출력하는 것이 정답지의 코드이나 연재님의 답안과 결과값은 동일합니다.

2-12, 4. 2-12번에서 그래프가 제대로 그려지지 않았는데, 같은 데이터셋을 사용하는 4번에서 clusterDF 데이터프레임을 출력했을 때에도 NaN값인 것으로 보아 x에 문제가 생긴게 아닐까 싶습니다. x 확인해보고 다시 풀어보시면 좋을 것 같아요.

5-2. 정답지입니다. 발표자료 50페이지 참고하시면 될 것 같아요.
from sklearn.mixture import GaussianMixture

n_components = np.arange(1, 16)
models = [GaussianMixture(n, random_state=33).fit(X_train_pca) for n in n_components]

aics = [model.aic(X_train_pca) for model in models]
bics = [model.bic(X_train_pca) for model in models]

plt.plot(n_components, aics, 'rs--')
plt.plot(n_components, bics, 'bo-')
plt.legend(['AIC', 'BIC'])
plt.show()

5-3. 정답지입니다.
gmm = GaussianMixture(n_components=3, random_state=33)
gmm.fit(X_train_pca)
print(gmm.converged_)

데이터 종류도 많고 복잡한 과제였는데 열심히 푸시느라 고생 많으셨습니다!!
